xcm/src/v0/mod.rs

```diff
@@ -203,6 +203,53 @@ pub enum Xcm {
 		#[codec(compact)] sender: u32,
 		#[codec(compact)] recipient: u32,
 	},
+
+	/// A message to propose opening a channel on the relay-chain between the
+	/// sender para to the recipient para.
+	///
+	/// - `recipient`: The recipient in the to-be opened channel.
+	/// - `max_message_size`: The maximum size of a message proposed by the sender.
+	/// - `max_capacity`: The maximum number of messages that can be queued in the channel.
+	///
+	/// Safety: The message should originate directly from the sender para.
+	///
+	/// Kind: *Instruction*.
+	///
+	/// Errors:
+	HrmpInitOpenChannel {
+		#[codec(compact)] recipient: u32,
+		#[codec(compact)] max_message_size: u32,
+		#[codec(compact)] max_capacity: u32,
+	},
+
+	/// A message to accept opening a channel on the relay-chain.
+	///
+	/// - `sender`: The sender in the to-be opened channel.
+	///
+	/// Safety: The message should originate directly from the recipient para.
+	///
+	/// Kind: *Instruction*.
+	///
+	/// Errors:
+	HrmpAcceptOpenChannel {
+		#[codec(compact)] sender: u32,
+	},
+
+	/// A message to close a channel on the relay-chain.
+	///
+	/// - `sender`: The sender in the to-be closed channel.
+	/// - `recipient`: The recipient in the to-be closed channel.
+	///
+	/// Safety: The message should originate directly from the sender or
+	/// recipient para (either member of the channel).
+	///
+	/// Kind: *Instruction*.
+	///
+	/// Errors:
+	HrmpCloseChannel {
+		#[codec(compact)] sender: u32,
+		#[codec(compact)] recipient: u32,
+	},
```

Comment:
Recalling an open channel request is upcoming in https://github.com/paritytech/polkadot/pull/3543

I think it's worth to reconsider these messages in the light of that PR.

Reply:
I don't see the open channel request XCM in that PR (and no new XCM message variants).

Is there a plan to add XCM variants for (1) request open channel (2) request close channel (3) accept open channel ?